packages/noir-ethereum-api/src/noir/oracles/accountOracles.ts

```diff
@@ -19,19 +20,44 @@ export interface AccountWithProof {
   depth: string;
 }
 
-export function serializeAccountWithProof(account: AccountWithProof): ForeignCallOutput[] {
-  return [account.balance, account.codeHash, account.nonce, account.key, account.value, account.proof, account.depth];
-}
-
 export const getAccountOracle = async (client: PublicClient, args: string[][]): Promise<ForeignCallOutput[]> => {
+  const { blockNumber, address } = parseNoirGetAccountArguments(args);
+  const accountProof: GetProofReturnType = await getAccountProof(client, address, blockNumber);
+  const accountWithProof = encodeAccount(accountProof);
+  return serializeAccountWithProof(accountWithProof);
+};
+
+export function parseNoirGetAccountArguments(args: string[][]): {
+  blockNumber: bigint;
+  address: Hex;
+} {
   assert(args.length === 2, 'get_account requires 2 arguments');
-  assert(args[0].length === 1, 'get_account first argument must be a block number');
-  assert(args[1].length === 42, 'get_account second argument must be an address');
-  const address = decodeHexAddress(args[1]);
-  const account = await client.getProof({ address, storageKeys: [] });
 
-  return [encodeField(account.balance), encodeHex(account.codeHash), encodeField(account.nonce)];
-};
+  const [noirBlockNumber, noirAddress] = args;
+
+  assert(noirBlockNumber.length === 1, 'get_account first argument must be an array of length 1');
+  assert(isHex(noirBlockNumber[0]), 'get_account first argument must be a hex value');
+
+  assert(noirAddress.length === 42, 'get_account second argument must be an address');
+  assert(all(isHex)(noirAddress), 'get_account second argument must be an array of hex string values');
+
+  const blockNumber: bigint = decodeField(noirBlockNumber[0]);
+  const address: Hex = decodeHexAddress(noirAddress);
+
+  return { blockNumber, address };
+}
+
+export async function getAccountProof(
+  client: PublicClient,
+  address: Hex,
+  blockNumber: bigint
+): Promise<GetProofReturnType> {
+  return (await client.getProof({ address, storageKeys: [], blockNumber })) as GetProofReturnType;
+}
```

Comment:
Why separate function that only calls one other function, why not inline it?

Reply:
ok, inlined